addons/html_editor/static/src/main/media/image_post_process_plugin.js

```diff
@@ -0,0 +1,437 @@
+    static id = "imagePostProcess";
+    static shared = ["processImage"];
+
+    /**
+     * Applies data-attributes modifications to an img tag and returns a dataURL
+     * containing the result. This function does not modify the original image.
+     *
+     * @param {HTMLImageElement} img the image to which modifications are applied
+     * @param {Object} newDataset an object containing the modifications to apply
+     * @returns {string} dataURL of the image with the applied modifications
+     */
+    async processImage(img, newDataset = {}) {
+        const processContext = {};
+        Object.assign(newDataset, await loadImageInfo(img));
+        for (const cb of this.getResource("process_image_warmup_handlers")) {
+            const addedContext = await cb(img, newDataset);
+            if (addedContext) {
+                if (addedContext.newDataset) {
+                    Object.assign(newDataset, addedContext.newDataset);
+                }
+                Object.assign(processContext, addedContext);
+            }
+        }
+
+        const data = Object.assign(
+            {
+                glFilter: "",
+                filter: "#0000",
+                quality: "75",
+                forceModification: false,
+            },
+            img.dataset,
+            newDataset
+        );
+        let {
+            width,
+            height,
+            resizeWidth,
+            quality,
+            filter,
+            mimetype,
+            originalSrc,
+            glFilter,
+            filterOptions,
+            forceModification,
+            aspectRatio,
+        } = data;
+
+        const { postProcessCroppedCanvas, perspective, getHeight } = processContext;
+
+        [width, height, resizeWidth] = [width, height, resizeWidth].map((s) => parseFloat(s));
+        quality = parseInt(quality);
+        // todo: maybe it should be in params and not saved in the dataset as
+        // this information could be inferred from x/y/width/height dataset
+        // properties.
+        aspectRatio = aspectRatio ? getAspectRatio(aspectRatio) : 0;
+
+        // Skip modifications (required to add shapes on animated GIFs).
+        if (isGif(mimetype) && !forceModification) {
+            const url = await loadImageDataURL(originalSrc);
+            return () => updateImageAttributes(url);
+        }
+
+        // Crop
+        const container = document.createElement("div");
+        const originalImg = await loadImage(originalSrc);
+        // loadImage may have ended up loading a different src (see: LOAD_IMAGE_404)
+        originalSrc = originalImg.getAttribute("src");
+        container.appendChild(originalImg);
+        const cropper = await activateCropper(originalImg, aspectRatio, data);
+        const croppedCanvas = cropper.getCroppedCanvas(width, height);
+        cropper.destroy();
+        const processedCanvas = (await postProcessCroppedCanvas?.(croppedCanvas)) || croppedCanvas;
```

Comment:
Why ?

Reply:
Why about what?
1. We re-use the logic of the cropper to crop an image.
If an image has some crop attributes (data-x/y/height/width/aspectRatio), the crop will be performed.
You have to understand that for any modification of an image, all the transformation are re-applied from the original src.

2. `postProcessCroppedCanvas` will be used by the svg plugin because some svg shape require to crop the cropped image.